packages/notebook-extension/src/tool-widgets/activeCellToolWidget.tsx

```diff
@@ -0,0 +1,110 @@
+ */
+
+import React from 'react';
+import { FieldProps } from '@rjsf/utils';
+import { IEditorLanguageRegistry } from '@jupyterlab/codemirror';
+import { INotebookTracker, NotebookTools } from '@jupyterlab/notebook';
+import { ISharedText } from '@jupyter/ydoc';
+import { PanelLayout, Widget } from '@lumino/widgets';
+import { CodeCellModel, ICellModel, InputPrompt } from '@jupyterlab/cells';
+import { Debouncer } from '@lumino/polling';
+
+namespace Private {
+  /**
+   * Custom active cell field options.
+   */
+  export interface IOptions {
+    /**
+     * The tracker to the notebook panel.
+     */
+    tracker: INotebookTracker;
+
+    /**
+     * Editor languages registry
+     */
+    languages: IEditorLanguageRegistry;
+  }
+}
+
+/**
+ * The active cell field, displaying the first line and execution count of the active cell.
+ *
+ * ## Note
+ * This field does not work as other metadata form fields, as it does not update metadata.
+ */
+export class ActiveCellTool extends NotebookTools.Tool {
+  constructor(options: Private.IOptions) {
+    super();
+    const { languages } = options;
+    this._tracker = options.tracker;
+
+    this.addClass('jp-ActiveCellTool');
+    this.layout = new PanelLayout();
+
+    this._inputPrompt = new InputPrompt();
+    (this.layout as PanelLayout).addWidget(this._inputPrompt);
+
+    // First code line container
+    const node = document.createElement('div');
+    node.classList.add('jp-ActiveCell-Content');
+    const container = node.appendChild(document.createElement('div'));
+    const editor = container.appendChild(document.createElement('pre'));
+    container.className = 'jp-Cell-Content';
+    this._editorEl = editor;
+    (this.layout as PanelLayout).addWidget(new Widget({ node }));
+
+    const update = async () => {
+      this._editorEl.innerHTML = '';
+      if (this._cellModel?.type === 'code') {
+        this._inputPrompt.executionCount = `${
+          (this._cellModel as CodeCellModel).executionCount ?? ''
+        }`;
+        this._inputPrompt.show();
+      } else {
+        this._inputPrompt.executionCount = null;
+        this._inputPrompt.hide();
+      }
+
+      if (this._cellModel) {
+        await languages.highlight(
+          this._cellModel.sharedModel.getSource().split('\n')[0],
+          languages.findByMIME(this._cellModel.mimeType),
+          this._editorEl
+        );
+      }
+    };
+
+    this._refreshDebouncer = new Debouncer(update, 150);
+  }
+
+  render(props: FieldProps): JSX.Element {
+    const activeCell = this._tracker.activeCell;
+    if (activeCell) this._cellModel = activeCell?.model || null;
+    (this._cellModel?.sharedModel as ISharedText).changed.connect(
+      this.refresh,
+      this
+    );
+    this._cellModel?.mimeTypeChanged.connect(this.refresh, this);
+    this.refresh()
+      .then(() => undefined)
+      .catch(() => undefined);
+    return (
+      <div className="cell-tool">
```

Comment:
Should all caps global const instead of inline CSS class string be used here?

Please see `shell.ts` for reference:
https://github.com/jupyterlab/jupyterlab/blob/82e1a74b61a41fbe5d65e8542053b749cdfe6ea0/packages/application/src/shell.ts#L39